app/models/site_configuration.rb

```diff
@@ -3,14 +3,66 @@ class SiteConfiguration < ApplicationRecord
   before_create :ensure_single_record
   validate :validate_help_link_items
 
+  has_many :revision_reasons
+
+  accepts_nested_attributes_for :revision_reasons, allow_destroy: true
+  validate :max_undiscarded_revision_reasons
+  validate :max_revision_reasons
+
   HELP_LINK_KEYS = %w[get_started_link_item best_practices_link_item dictionary_link_item user_guide_link_item]
 
   def self.instance
     first_or_create
   end
 
+  def revision_reasons_attributes=(attributes)
+    attributes.each do |attribute, _|
+      if attribute["_destroy"] == true
+        # Mark the record for destruction
+        self.revision_reasons.find(attribute["id"])&.discard if attribute["id"].present?
```

Comment:
the `find` method will throw an error if the `revision_reasons` is not found. If this is not intentional change the method to `find_by`